### PR TITLE
Create invitations

### PR DESCRIPTION
**Pull request by MaksimHamko**
## Description

You can invite people to organisations using their email

## Type of PR

- Feature

## Related Issues/Tickets
https://github.com/aau-giraf/weekplan/issues/82

## Checklist

- [x] Completed the user story described by the issue (if applicable)
- [x] Code follows the code guidelines
- [x] Tests have been added/updated (if applicable)
- [x] Documentation has been updated (if applicable)

### CODE REVIEW

File: apis/invitationAPI.ts [hunk 1]
```diff
@@ -24,3 +24,16 @@ export const acceptInvitationRequest = async (
   });
   if (!res.ok) throw new Error("Kunne ikke acceptere invitation");
 };
+
+export const createInvitationRequest = async (orgId: number, receiverEmail: string, senderId: string) => {
+  const newInvitation = {organizationId: orgId, receiverEmail, senderId};
+  console.log(newInvitation);
```
Comment (MertzAndreas) [inline apis/invitationAPI.ts:30]: eww

File: hooks/useInvitation.ts [hunk 2]
```diff
@@ -43,5 +44,32 @@ export default function useInvitation() {
...
+      const previousInvitations = queryClient.getQueryData(queryKey);
+
+      return { previousInvitations };
+    },
+
+    onError: (_error, _variables, context) => {
+      if (context?.previousInvitations) {
+        queryClient.setQueryData(queryKey, context.previousInvitations);
+      }
+    }
```
Comment (MertzAndreas) [inline hooks/useInvitation.ts:69]: Either these functions should be deleted set the cache of the organisation which has invited this user.

**Review by MertzAndreas — CHANGES_REQUESTED** [review]
Please set up prettier and go through the files 😮‍💨

**Review by Azzi132 — APPROVED** [review]
Looks good!

File: app/create-invitation.tsx
Comment (MertzAndreas) [inline app/create-invitation.tsx:1]: Migrate this to the new form API and components

**Review by MertzAndreas — CHANGES_REQUESTED** [review]

File: app/create-invitation.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+import { colors, ScaleSize } from "../utils/SharedStyles";
+import useInvitation from "../hooks/useInvitation";
+import { useLocalSearchParams } from "expo-router";
+import { useAuthentication } from "../providers/AuthenticationProvider";
+import { useToast } from "../providers/ToastProvider";
+
+const CreateInvitationPage: React.FC = () => {
+  const [email, setEmail] = useState("");
+  const { createInvitation } = useInvitation();
+  const createInv = createInvitation;
```
Comment (MertzAndreas) [inline app/create-invitation.tsx:18]: What's this?

**Review by MertzAndreas — APPROVED** [review]

File: app/create-invitation.tsx (hunk 3 above)
Comment (MertzAndreas) [inline app/create-invitation.tsx:18]: Why not just use as it?

**Review by MertzAndreas — CHANGES_REQUESTED** [review]

**Review by MertzAndreas — APPROVED** [review]

**Pull request merged**